### PR TITLE
Add --no-eval option to merge

### PR DESCRIPTION
**Pull request by jszroberto**
Created pull request so we can discuss here the implementation details.

### CODE REVIEW

**Comment by jszroberto** [comment]
@jhunt  I think we are seeing 2 main use cases here: 

1) Merge things into  'main' template --> We don't want to keep the merge operators. We actually want to execute them and see that they work. 

2) Merge several stubs into a big stub, for which we would need that all operators remain in the manifest. 

I think it makes sense to limit this pull request to 1)  which is the one you are also thinking.

For 1)  do you think we should execute the prunes?

**Comment by jhunt** [comment]
I think if you were asked to prune, then yes, we should handle those.  Same thing goes for `--cherry-pick`

**Comment by geofffranks** [comment]
Can you add tests for the pruning + cherrypicking with SkipEval?

I'm curious if it differentiates `--prune stuff` from `dont_eval: (( prune ))`

**Comment by jszroberto** [comment]
@geofffranks yes, it is in my TODO.

**Comment by jszroberto** [comment]
@geofffranks please, take a look now :)

**Comment by jszroberto** [comment]
Strange, tests pass locally for me.

**Comment by jszroberto** [comment]
Ok, it's after changes introduced in the diff. Taking a look

**Comment by jszroberto** [comment]
The problematic commit comes from https://github.com/geofffranks/spruce/pull/197/files

**Comment by jhunt** [comment]
I would disagree with the test on line 295.  If we `--skip-eval`, the `(( prune ... ))` forms should pass-through unevaluated.  If we specify `--prune`, those paths should be pruned (the next test).

I think you just need to modify your test so that it expects the `(( prune ... ))` form to persist, and then we're good to go.

**Comment by jszroberto** [comment]
@jhunt  done

File: cmd/spruce/main_test.go [hunk 1]
```diff
@@ -250,6 +250,53 @@ map:
 			So(stderr, ShouldEqual, "")
 		})
 
+		Convey("Should not evaluate spruce logic appart from (( prune )) when --no-eval", func() {
```
Comment (geofffranks) [inline cmd/spruce/main_test.go]: Can this test case name be updated?  It looks like the test is ensuring that the prune operator and other operators are not evaluated

Comment (jszroberto) [inline cmd/spruce/main_test.go]: It's what it supposed to happen. If you specify --no-eval, then they prun operator and other operators aren't evaluated... 

not sure, what you expect here

Comment (jhunt) [inline cmd/spruce/main_test.go]: Can you change the name of the test case?  It seems to imply that `(( prune ... ))` _should_ be processed.

Comment (jszroberto) [inline cmd/spruce/main_test.go]: ahh, right

Comment (jszroberto) [inline cmd/spruce/main_test.go]: now

**Pull request merged**